cypress/utils/apps/query_enhancements/saved_queries.js

```diff
@@ -1,207 +0,0 @@
- * @typedef {Object} SavedSearchTestConfig
- * @property {string} dataset - the dataset name to use
- * @property {QueryEnhancementDataset} datasetType - the type of dataset
- * @property {QueryEnhancementLanguage} language - the name of query language as it appears in the dashboard app
- * @property {string} apiLanguage - the name of query language as recognized by OpenSearch API
- * @property {string} saveName - the name to use when saving the saved query
- * @property {string} testName - the phrase to add to the test case's title
- * @property {boolean} filters - whether the language supports filtering
- * @property {boolean} histogram - whether the language supports histogram
- * @property {boolean} selectFields - whether the language supports selecting fields to view data
- * @property {boolean} sort - whether the language supports sorting by fields
- * @property {string} queryString - the query to use for saved query associated with the language
- * @property {number|undefined} hitCount - the hitCount of the applied query config, if relevant
- * @property {[[number,string]]|*[]} sampleTableData - an array of some table data to test against to ensure that sorting is working as expected
- */
-
-/**
- * Returns the SavedSearchTestConfig for the provided dataset, datasetType, and language
- * @param {string} dataset - the dataset name
- * @param {QueryEnhancementDataset} datasetType - the type of the dataset
- * @param {QueryEnhancementLanguageData} language - the relevant data for the query language to use
- * @returns {SavedSearchTestConfig}
- */
-const generateTestConfiguration = (dataset, datasetType, language) => {
-  const baseConfig = {
-    dataset,
-    datasetType,
-    language: language.name,
-    apiLanguage: language.apiName,
-    saveName: `${language.name}-${datasetType}`,
-    testName: `${language.name}-${datasetType}`,
-    ...language.supports,
-  };
-
-  return {
-    ...baseConfig,
-    queryString: getQueryString(dataset, language.name),
-    hitCount: getExpectedHitCount(datasetType, language.name),
-    sampleTableData: getSampleTableData(datasetType, language.name),
-  };
-};
-
-/**
- * Returns an array of test configurations for every query language + dataset permutation
- * @returns {SavedSearchTestConfig[]}
- */
-export const generateAllTestConfigurations = () => {
-  return Object.values(DatasetTypes).flatMap((dataset) =>
-    dataset.supportedLanguages.map((language) => {
-      let datasetToUse;
-      switch (dataset.name) {
-        case DatasetTypes.INDEX_PATTERN.name:
-          datasetToUse = INDEX_PATTERN_WITH_TIME;
-          break;
-        case DatasetTypes.INDEXES.name:
-          datasetToUse = INDEX_WITH_TIME_1;
-          break;
-        default:
-          throw new Error(
-            `generateAllTestConfigurations encountered unsupported dataset: ${dataset.name}`
-          );
-      }
-      return generateTestConfiguration(datasetToUse, dataset.name, language);
-    })
-  );
-};
```

Comment:
All of these utilities already existed in `saved_search.js` util file. So I renamed the `saved_search.js` file to `saved.js` and changed the names/comments of some of the functions so that it is clear that it is to be used by both `savedSearch` and `savedQueries`